### DISCUSSION TITLE
Task ansible:prepare fails on install of nbd/rbd

### DISCUSSION THREAD

**Discussion by casmith** [onedr0p/cluster-template#842] [General]
I'm trying to run this on a pair of raspberry pi 4 (8gb ram) running ubuntu 22.04, and when I run task ansible:prepare I am met with the following error:

`failed: [kube1] (item=nbd) => changed=false 
  ansible_loop_var: item
  item: nbd
  msg: |-
    modprobe: FATAL: Module nbd not found in directory /lib/modules/5.15.0-1033-raspi
  name: nbd
  params: ''
  rc: 1
  state: present
  stderr: |-
    modprobe: FATAL: Module nbd not found in directory /lib/modules/5.15.0-1033-raspi
  stderr_lines: <omitted>
  stdout: ''
  stdout_lines: <omitted>
failed: [kube2] (item=nbd) => changed=false 
  ansible_loop_var: item
  item: nbd
  msg: |-
    modprobe: FATAL: Module nbd not found in directory /lib/modules/5.15.0-1033-raspi
  name: nbd
  params: ''
  rc: 1
  state: present
  stderr: |-
    modprobe: FATAL: Module nbd not found in directory /lib/modules/5.15.0-1033-raspi
  stderr_lines: <omitted>
  stdout: ''
  stdout_lines: <omitted>
ok: [kube1] => (item=overlay)
ok: [kube2] => (item=overlay)
failed: [kube1] (item=rbd) => changed=false 
  ansible_loop_var: item
  item: rbd
  msg: |-
    modprobe: FATAL: Module rbd not found in directory /lib/modules/5.15.0-1033-raspi
  name: rbd
  params: ''
  rc: 1
  state: present
  stderr: |-
    modprobe: FATAL: Module rbd not found in directory /lib/modules/5.15.0-1033-raspi
  stderr_lines: <omitted>
  stdout: ''
  stdout_lines: <omitted>
failed: [kube2] (item=rbd) => changed=false 
  ansible_loop_var: item
  item: rbd
  msg: |-
    modprobe: FATAL: Module rbd not found in directory /lib/modules/5.15.0-1033-raspi
  name: rbd
  params: ''
  rc: 1
  state: present
  stderr: |-
    modprobe: FATAL: Module rbd not found in directory /lib/modules/5.15.0-1033-raspi
  stderr_lines: <omitted>
  stdout: ''
  stdout_lines: <omitted>
`

I tried removing the modprobe and chmod for these 2 modules from the cluster-prepare playbook, which gets me through the installation, but then I have other issues with cidium not running. I'm not sure if it is related. 

My kernel is: 

Linux kube2 5.15.0-1033-raspi #36-Ubuntu SMP PREEMPT Thu Jun 22 08:10:31 UTC 2023 aarch64 aarch64 aarch64 GNU/Linux

**Comment by onedr0p** [comment]
Since I dropped support for Ubuntu in favor of Debian 12 you'll need to add the package `linux-modules-extra-raspi`, or if not using rook-ceph remove nbd and rbd like you already did.

Although none of these things matter to Cilium.

**Reply by casmith** [reply]
Thanks a bunch! Adding `linux-modules-extra-raspi` worked great.
